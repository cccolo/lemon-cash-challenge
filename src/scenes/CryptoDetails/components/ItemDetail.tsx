import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

type Props = {
  title: string;
  description: string | number;
};

export const ItemDetail: React.FC<Props> = ({title, description}) => {
  return (
    <View style={styles.table}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
      <View style={styles.valueContainer}>
        <Text style={styles.value}>{description}</Text>
      </View>
    </View>
  );
};

// TODO:
const styles = StyleSheet.create({
  table: {
    flexDirection: 'row',
    marginTop: 18,
  },
  titleContainer: {
    flex: 1,
  },
  title: {
    color: 'rgb(0, 240, 104)',
    fontFamily: 'NeueMachina-Regular',
    fontSize: 12,
  },
  valueContainer: {
    flex: 1,
    alignItems: 'flex-end',
  },
  value: {
    color: 'white',
    fontFamily: 'NeueMachina-Regular',
    fontSize: 12,
  },
});
