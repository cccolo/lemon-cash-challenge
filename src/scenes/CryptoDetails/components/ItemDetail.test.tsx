import React from 'react';
import {render} from '@testing-library/react-native';
import {ItemDetail} from './ItemDetail';

describe('ItemDetail', () => {
  it('should render correctly and match snapshot', () => {
    const title = 'Test Title';
    const description = 'Test Description';

    const {toJSON} = render(
      <ItemDetail title={title} description={description} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render title and description correctly', () => {
    const title = 'Test Title';
    const description = 'Test Description';

    const {getByText} = render(
      <ItemDetail title={title} description={description} />,
    );

    expect(getByText(title)).toBeTruthy();
    expect(getByText(description)).toBeTruthy();
  });

  it('should render description as a number correctly', () => {
    const title = 'Test Title';
    const description = 123;

    const {getByText} = render(
      <ItemDetail title={title} description={description} />,
    );

    expect(getByText(title)).toBeTruthy();
    expect(getByText(description.toString())).toBeTruthy();
  });
});
