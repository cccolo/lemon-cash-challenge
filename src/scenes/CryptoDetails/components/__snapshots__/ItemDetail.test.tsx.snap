// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail should render correctly and match snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginTop": 18,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(0, 240, 104)",
          "fontFamily": "NeueMachina-Regular",
          "fontSize": 12,
        }
      }
    >
      Test Title
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(255, 255, 255)",
          "fontFamily": "NeueMachina-Regular",
          "fontSize": 12,
        }
      }
    >
      Test Description
    </Text>
  </View>
</View>
`;
